### PR TITLE
Add stubs for Dart 2.0 changes to Iterable and Set

### PR DESCRIPTION
**Pull request by keertip**
@kevmoo

### CODE REVIEW

File: lib/src/css_class_set.dart [hunk 1]
```diff
@@ -327,4 +328,36 @@ abstract class CssClassSetImpl implements CssClassSet {
    * This is intended to be overridden by specific implementations.
    */
   void writeClasses(Set<String> s);
+
```
Comment (leafpetersen) [inline lib/src/css_class_set.dart]: I think you're missing the signature change to singleWhere.

Comment (keertip) [reply lib/src/css_class_set.dart]: done.

File: lib/src/list_proxy.dart [hunk 2]
```diff
@@ -111,4 +111,63 @@ class ListProxy<E> extends IterableBase<E> implements List<E> {
 
   void fillRange(int start, int end, [E fillValue]) =>
       _list.fillRange(start, end, fillValue);
+
```
[the remark this answers is not in the document]
Comment (leafpetersen) [reply lib/src/list_proxy.dart]: Ah, then in that case no.

**Comment by keertip** [comment]
PTAL

File: lib/src/css_class_set.dart [hunk 3]
```diff
@@ -33,7 +33,7 @@ class ElementCssClassSet extends CssClassSetImpl {
 }
 
 /** A Set that stores the CSS class names for an element. */
-abstract class CssClassSet implements Set<String> {
+abstract class CssClassSet extends SetBase<String> {
```
Comment (leafpetersen) [inline lib/src/css_class_set.dart]: I'm not sure these are in the right places, but I could be wrong.  I think it looks like:

abstract class CssClassSet ....

abstract class CssClassSetImpl implements CssClassSet

class ElementCssClassSet extends CssClassSetImpl

So I think the new methods needs to go on either CssClassSetImpl or ElementCssClasSet

I think best would be making CssClassSetImpl extend SetBase<String> implements CssClassSet?

Or am I confused?

Comment (keertip) [reply lib/src/css_class_set.dart]: you are right.

Kevin has a [PR](https://github.com/dart-lang/html/pull/58) which cleans up the package along with these changes. So going to go with that and discard this one.

**Pull request closed without merging**